### PR TITLE
This fixes the TLS handshake on Windows.

### PR DESCRIPTION
**Pull request by phillipsj**
* https://github.com/rancher/windows/issues/161

### CODE REVIEW

File: pkg/k8splan/watcher.go [hunk 1]
```diff
@@ -365,7 +370,33 @@ func (w *watcher) updateSecret(core corecontrollers.Interface, secret *v1.Secret
...
+
+	config.Transport = utilnet.SetTransportDefaults(&http.Transport{
+		DialTLS: func(network, addr string) (net.Conn, error) {
+			conn, err = tls.Dial(network, addr, tlsConfig)
+			return conn, err
+		},
+	})
+	config.WrapTransport = transportConfig.WrapTransport
+	config.Dial = transportConfig.Dial
+	// Overwrite TLS-related fields from config to avoid collision with
```
Comment (jakefhyde) [inline pkg/k8splan/watcher.go:396]: Can you add some more context regarding why we had to do this (i.e. specifically call out how go behaves on Windows vs. Linux regarding transports)?

**Review by jakefhyde — APPROVED** [review]
LGTM with 1 nit

**Review by nickgerace — APPROVED** [review]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

**Review by rosskirkpat — APPROVED** [review]
LGTM 🚀🚀🚀

**Review by Oats87 — APPROVED** [review]
LGTM

**Pull request merged**